src/components/chip/chip.tsx

```diff
@@ -123,6 +124,13 @@ export class Chip {
     @Prop({ reflect: true })
     public removable = false;
 
+    /**
+     * Set to `filter` to render the chip with a distinct style
+     * suitable for visualizing filters.
+     */
+    @Prop({ reflect: true })
+    public type: 'filter';
+
```

Comment:
Almost all the props above exist in src/components/chip-set/chip.types.ts , which are used in `limel-chip-set` today. However, I think it's better to have the props of this new component inside it.

Also, in src/components/chip-set/chip.types.ts, there is a `value` prop, which I don't really know why it is needed.

Reply:
Hmm… yeah… I'm slightly conflicted… I'm pretty confident we'll want a `Chip` type* in the future as well, because we need to be able to pass around arrays of them and stuff. And since it's annoying to have to define the same thing twice, and keep stuff in sync, that would seem like an argument for having the limel-chip component accept a Chip. And when used in something like the chip-set, it's surely going to be more convenient to just pass the `Chip` straight into the component. But when used stand-alone, it's probably nicer not to have to pass in a `Chip`, but pass the properties separately… 🤔

Or, maybe… perhaps that's just when we think about using limel-chip in the kind of way that you use a button. Which is exactly the way we _don't_ want people to use it. If you actually use it to represent entities, you're much more likely to have an array of things, and then you can just map those things to Chips, and, most of all, you can have a single listener for changes, that expects an updated `Chip`, rather than have one listener for each property.

I might flip-flop on this over the course of reviewing this, but for now, I'm leaning heavily towards accepting a `Chip`, rather then separate properties.

(* pedant's corner: the current one is an interface, I'll call it "type" when I'm talking about something that defines a data structure, as I tend to think of "interface" as "something that can be implemented")

Reply:
> using limel-chip in the kind of way that you use a button. Which is exactly the way we don't want people to use it.

I absolutely want people to be able to use it as an stand-alone component too. But if they need a chip-set, it's for sure better to use `limel-chip-set` to get more relevant features.

Reply:
My feeling is that the consumer component (e.g. chip-set) can import the stuff from `chip.types.ts`, but that should perhaps be located in this new component's folder instead.

Reply:
@jgroth @vicgeralds Can you chime in here? I'm thinking that, if you use limel-chip stand-alone (the way it's used in the examples), it's a minor inconvenience to have to create a `Chip` object to pass to `limel-chip`. But I feel like _most_ times someone would use a chip (and when we want to use it in a future version of `limel-chip-set`) you would have an array of `Chip`s. Again, having to apply those values as individual attributes is really only a minor inconvenience, but it's when it comes to event handlers that I'm getting a little worried that separate attributes might make things a major pain in the butt:

- If we pass a `Chip` as a single property, then we'd presumably pass that chip along with any event that happens. Any change to the chip's status could either be reflected in the `Chip` properties (we might have a property `Chip.selected`, for example), or by having separate events (`chipSelected` and `chipDeselected`, for example). You could then listen for any event on some container, rather than hooking up one listener for each event on each chip, because you would know _which_ chip the event applies to—the affected chip is attached to the event!
- If, on the other hand, we use separate attributes, our convention is to pass the value of the property that changed. So the event for a chip being selected would just be `selected`, with a value of `true` or `false`. We could of course choose to depart from that convention, but that would be unexpected and presumably confusing (including to our future selves, I believe). So, assuming we keep to our convention, we need to know from _which chip_ an event came. Which means either hooking up separate listeners for each chip, which isn't nice performance-wise, or inspecting the event to find out which DOM element was clicked, and from there figure out which chip that correlates to. "…and there shall be much wailing and gnashing of teeth…" 🙈

Reply:
I think having a single `chip` property is a bit annoying and not how we usually have done things. The chip component should probably implement the `Chip` interface to ensure the component and the interface are in sync

Reply:
Having an object as a single property makes it more difficult/annoying to detect changes as well. If the passed chip object isn't strictly equal to the old chip object, that will cause re-rendering even if there's no real changes.

> we need to know from _which chip_ an event came. Which means either hooking up separate listeners for each chip, which isn't nice performance-wise, or inspecting the event to find out which DOM element was clicked, and from there figure out which chip that correlates to. 

We don't make it any easier to identify which input field a change event originated from. Why should we be concerned then about chips?

I don't think the individual component should need to worry about identifying itself among a collection of others.

Reply:
> I think having a single `chip` property is a bit annoying and not how we usually have done things.

> We don't make it any easier to identify which input field a change event originated from. Why should we be concerned then about chips?

Alright, then that's settled! Good thing I asked! 😄

Thank you! 🙏